src/redux/components/MessageList.jsx

```diff
@@ -121,17 +121,17 @@ class MessageListItem extends React.Component {
 MessageList.defaultProps = {};
 
 MessageList.propTypes = {
-  onCommitMessage: PropTypes.func,
+  onCommitMessage: PropTypes.func.isRequired,
+  messages: PropTypes.array.isRequired,
   message: PropTypes.object
 };
```

Comment:
Maybe it's time to remove default props and prop types, since both are covered with modern typescript?

Reply:
Rewriting `MessageList` is part of the requirements. I'd say lets leave that to the candidate.

Also, fun fact: with `"allowJs": true` set, TS is capable of inferring (to some extent) the type of props from `propTypes`